src/instructlab/model/evaluate.py

```diff
@@ -258,13 +258,18 @@ def launch_server(
     if backend == backends.VLLM:
         eval_serve.vllm.vllm_args.extend(["--served-model-name", model_name])
         if gpus:
-            # Don't override from vllm_args
-            if "--tensor-parallel-size" not in eval_serve.vllm.vllm_args:
-                eval_serve.vllm.vllm_args.extend(["--tensor-parallel-size", str(gpus)])
-            else:
-                click.echo(
-                    "Ignoring --gpus with --tensor-parallel-size configured in serve vllm_args"
+            # Warn when overriding from vllm_args
+            tps_prefix = "--tensor-parallel-size"
+            if any(
+                s == tps_prefix or s.startswith(tps_prefix + "=")
+                for s in eval_serve.vllm.vllm_args
+            ):
```

Comment:
It's a common problem. Let's make this a helper function in `utils`. This should do the trick:

```python
def split_arg_options(args: list[str]) -> set[str]:
    return set(arg.split("=", 1)[0]) for arg in args of arg.startswith("--"))
```

Reply:
I was hoping to use contains_argument from https://github.com/instructlab/instructlab/pull/1636/files once it goes in.

Reply:
@danmcp I think this one might go in before #1636

Reply:
> @danmcp I think this one might go in before #1636

If it does, I'll can do a follow-on.  Or I can create the function in utils if that's preferred.